python_modules/libraries/dagster-airlift/kitchen-sink/kitchen_sink/airflow_dags/migrated_asset_has_retries.py

```diff
@@ -0,0 +1,62 @@
+from pathlib import Path
+from typing import Any, Mapping
+
+from airflow import DAG
+from airflow.operators.python import PythonOperator
+from dagster._time import get_current_datetime_midnight
+from dagster_airlift.in_airflow import DefaultProxyTaskToDagsterOperator, proxying_to_dagster
+from dagster_airlift.in_airflow.dagster_run_utils import (
+    MAX_RETRIES_TAG,
+    RETRY_ON_ASSET_OR_OP_FAILURE_TAG,
+)
+from dagster_airlift.in_airflow.proxied_state import load_proxied_state_from_yaml
+
+
+def print_hello() -> None:
+    print("Hello")  # noqa: T201
+
+
+default_args = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "retries": 0,
+}
+
+
+# Normally this isn't needed, but we're trying to get away with not using a multi-process-safe run storage
+# to test behavior here.
+class SetDagsterRetryInfoOperator(DefaultProxyTaskToDagsterOperator):
+    def default_dagster_run_tags(self, context) -> Mapping[str, Any]:
+        tags = {**super().default_dagster_run_tags(context), MAX_RETRIES_TAG: "3"}
+        if self.get_airflow_dag_id(context).endswith("not_step_failure"):
+            tags[RETRY_ON_ASSET_OR_OP_FAILURE_TAG] = "false"
+        return tags
```

Comment:
i'm not totally sure what's going on here, what problems do you see without this?

Reply:
we're configuring the retry behavior as part of the airflow run instead of configuring it on the Dagster instance, since we're using a non-persistent run storage. You can't actually configure retry behavior for a non-persistent run storage (or at least couldn't when I added this). So this is a workaround to engage the retry system manually.

Reply:
To be honest I don't really want to go reup the test setup here and then deal with the rebase; if we can simplify here I'll do it in a followup